server.test.js

```diff
@@ -0,0 +1,135 @@
+const url = require('url')
+const { processRedirectString, getRedirect } = require('./server')
+
+describe('processRedirectString', () => {
+  it('reads the regex, replacement and code', () => {
+    const { regex, replace, code } = processRedirectString('^/foo /bar 418')
+
+    expect(regex).toBeInstanceOf(RegExp)
+    expect(regex.source).toEqual('^\\/foo')
+    expect(replace).toEqual('/bar')
+    expect(code).toEqual(418)
+  })
+  it('code defaults to 301', () => {
+    const { code } = processRedirectString('^/x /x')
+
+    expect(code).toEqual(301)
+  })
+  it('detects whether we are matching a pathname or the whole url', () => {
```

Comment:
convention is to have a single space between test in the existing tests (one single file probably) :)

Reply:
Sorry, I missed that. I'll add spaces.